cdap-app-templates/cdap-etl/cdap-data-streams/src/test/java/co/cask/cdap/datastreams/DataStreamsTest.java

```diff
@@ -89,6 +93,58 @@ public static void setupTest() throws Exception {
   }
 
   @Test
+  public void testSparkSink() throws Exception {
+    Schema schema = Schema.recordOf(
+      "test",
+      Schema.Field.of("id", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("name", Schema.of(Schema.Type.STRING))
+    );
+    List<StructuredRecord> input = new ArrayList<>();
+    StructuredRecord samuelRecord = StructuredRecord.builder(schema).set("id", "123").set("name", "samuel").build();
+    StructuredRecord jacksonRecord = StructuredRecord.builder(schema).set("id", "456").set("name", "jackson").build();
+    StructuredRecord dwayneRecord = StructuredRecord.builder(schema).set("id", "789").set("name", "dwayne").build();
+    StructuredRecord johnsonRecord = StructuredRecord.builder(schema).set("id", "0").set("name", "johnson").build();
+    input.add(samuelRecord);
+    input.add(jacksonRecord);
+    input.add(dwayneRecord);
+    input.add(johnsonRecord);
+
+    DataStreamsConfig etlConfig = DataStreamsConfig.builder()
+      .addStage(new ETLStage("source", MockSource.getPlugin(schema, input)))
+      .addStage(new ETLStage("sink", co.cask.cdap.etl.mock.spark.streaming.MockSink.getPlugin("sinktable")))
+      .addConnection("source", "sink")
+      .setBatchInterval("1s")
+      .build();
+
+    ApplicationId appId = NamespaceId.DEFAULT.app("sparksinkapp");
+    AppRequest<DataStreamsConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    SparkManager sparkManager = appManager.getSparkManager(DataStreamsSparkLauncher.NAME);
+    sparkManager.start();
+    sparkManager.waitForStatus(true, 10, 1);
+    final DataSetManager<KeyValueTable> outputManager = getDataset("sinktable");
+    final List<Integer> expectedIds = Arrays.asList(123, 456, 789, 0);
```

Comment:
seems like this should be a Set, in general, order is not guaranteed in pipelines.